smslibrary/src/main/java/com/eis/smslibrary/SMSHandler.java

```diff
@@ -196,12 +196,12 @@ public void removeReceivedListener() {
     }
 
     /**
-     * Helper function that gets the message content by using the pre-setup parser in {@link SMSMessageHandler}
+     * Helper function that gets the message content by using the pre-setup parser in {@link SMSMessageParser}
      *
      * @param message to get the data from
      * @return the data parsed from the message
      */
     private String getSMSContent(SMSMessage message) {
-        return SMSMessageHandler.getInstance().parseData(message);
+        return SMSMessageParser.getInstance(context.get()).parseData(message);
```

Comment:
Here you're adding a character to the text of the `message` to be sent (you're calling method `parseData()` from `DefaultSMSPartStrategy`), but you're not checking if it still fits `SMSMessage.MAX_MSG_TEXT_LEN`